File: FrontendVebProdavnica/src/app/shared/models/porudzbina.ts
export class PorudzbinaCreationDTO {
    datum: Date;
    status: string;
    iznos: number;
    korisnikID: number;

    constructor(
        datum: Date,
        status: string,
        iznos: number,
        korisnikID: number,
        
      ) {
        this.datum = datum;
        this.status = status;
        this.iznos = iznos;
        this.korisnikID = korisnikID;
      }
  }
  

  export class PorudzbinaDTO {
    porudzbinaID: number;
    datum: Date;
    status: string;
    iznos: number;
    korisnikID: number;
    paymentIntentId: string | null;
    clientSecret: string | null;

    constructor(
        porudzbinaID: number,
        datum: Date,
        status: string,
        iznos: number,
        korisnikID: number,
        paymentIntentId: string | null,
        clientSecret: string | null
      ) {
        this.porudzbinaID = porudzbinaID;
        this.datum = datum;
        this.status = status;
        this.iznos = iznos;
        this.korisnikID = korisnikID;
        this.paymentIntentId = paymentIntentId;
        this.clientSecret = clientSecret;
      }
  }
  

  export interface PorudzbinaPaymentDTO {
    porudzbinaID: number;
    datum: Date;
    status: string;
    iznos: number;
    korisnikID: number;
    paymentIntentId: string;
    clientSecret: string;
  }
  

File: VebProdavnica/VebProdavnicaDecijeIgracke/VPDecijeIgracke/VPDecijeIgracke/Models/PorudzbinaModel/Porudzbina.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using VPDecijeIgracke.Models.KorisnikModel;
using VPDecijeIgracke.Models.StavkaPorudzbineModel;
using System.Text.Json.Serialization;

namespace VPDecijeIgracke.Models.PorudzbinaModel
{
    public class Porudzbina
    {
        public Porudzbina()
        {
            Status = string.Empty;
        }

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int PorudzbinaID { get; set; }

        [Required(ErrorMessage = "Datum porudzbine je obavezan.")]
        public DateTime Datum { get; set; }

        [Required(ErrorMessage = "Status porudzbine je obavezan.")]
        [StringLength(20)]
        public string Status { get; set; }

        public decimal Iznos { get; set; }
        public string? PaymentIntentId { get; set; }
        public string? ClientSecret { get; set; }

        //strani kljucevi
        public int KorisnikID { get; set; }
        [ForeignKey("KorisnikID")]
        public Korisnik Korisnik { get; set; }

        [JsonIgnore]
        public ICollection<StavkaPorudzbine> StavkePorudzbine { get; set; }
    }
}
